server/api/build.go

```diff
@@ -301,30 +302,46 @@ func PostApproval(c *gin.Context) {
 		return
 	}
 
-	netrc, err := _remote.Netrc(user, repo)
-	if err != nil {
-		c.String(500, "failed to generate netrc file. %s", err)
+	if build, err = shared.UpdateToStatusPending(_store, *build, user.Login); err != nil {
+		c.String(http.StatusInternalServerError, "error updating build. %s", err)
 		return
 	}
 
-	if build, err = shared.UpdateToStatusPending(_store, *build, user.Login); err != nil {
-		c.String(500, "error updating build. %s", err)
+	var yamls []*remote.FileMeta
+	for _, y := range configs {
+		yamls = append(yamls, &remote.FileMeta{Data: y.Data, Name: y.Name})
+	}
+
+	startBuild(c, build, user, repo, yamls)
+}
+
+func startBuild(c *gin.Context, build *model.Build, user *model.User, repo *model.Repo, yamls []*remote.FileMeta) {
```

Comment:
Yeah, nice function this will be really helpful for other features like manual-triggers, cron, ... as well. Could we change it directly a bit so it doesn't require the gin-context argument and returns the build or an error.